app/javascript/modal.js

```diff
@@ -85,15 +85,14 @@ function openModal(html, confirmCallback, denyCallback) {
       return;
     }
 
+    modalElement.removeEventListener('click', confirmationEventDelegate);
+    contentElement.innerHTML = '';
+    containerElement.classList.remove(MODAL_SHOW_CLASS);
```

Comment:
Earlier uses of this modal utility always resulted in a page reload/change after success. In this case, there is no page reload and so we change the implementation here to always clear the modal.